lib/streamlit/runtime/pages_manager.py

```diff
@@ -51,7 +51,56 @@ def _on_pages_changed(_path: str) -> None:
                 glob_pattern="*.py",
                 allow_nonexistent=True,
             )
-            PagesManagerV1.is_watching_pages_dir = True
+            PagesStrategyV1.is_watching_pages_dir = True
+
+    def __init__(self, pages_manager: PagesManager, **kwargs):
+        self.pages_manager = pages_manager
+
+        if kwargs.get("setup_watcher", True):
+            PagesStrategyV1.watch_pages_dir(pages_manager)
+
+    # In MPA v1, there's no difference between the active hash
+    # and the page script hash.
+    def get_active_script_hash(self) -> PageHash:
+        return self.pages_manager.current_page_hash
+
+    def set_active_script_hash(self, _page_hash: PageHash):
+        raise NotImplementedError(
+            "Unable to set the active script hash in this strategy"
+        )
+
+    def get_active_script(
+        self, page_script_hash: PageHash, page_name: PageName
+    ) -> Optional[PageInfo]:
+        pages = self.pages_manager.get_pages()
+
+        if page_script_hash:
+            return pages.get(page_script_hash, None)
+        elif not page_script_hash and page_name:
+            # If a user navigates directly to a non-main page of an app, we get
+            # the first script run request before the list of pages has been
+            # sent to the frontend. In this case, we choose the first script
+            # with a name matching the requested page name.
+            return next(
+                filter(
+                    # There seems to be this weird bug with mypy where it
+                    # thinks that p can be None (which is impossible given the
+                    # types of pages), so we add `p and` at the beginning of
+                    # the predicate to circumvent this.
+                    lambda p: p and (p["page_name"] == page_name),
+                    pages.values(),
+                ),
+                None,
+            )
+
+        # If no information about what page to run is given, default to
+        # running the main page.
+        # Safe because pages will at least contain the app's main page.
+        main_page_info = list(pages.values())[0]
+        return main_page_info
+
+    def get_pages(self) -> dict[PageHash, PageInfo]:
+        return get_pages(self.pages_manager.main_script_path)
```

Comment:
nit: Would be worth renaming `source_util.get_pages()` because otherwise this can be slightly confusing